### ISSUE TITLE
Dealing with missing node features

### ISSUE THREAD

**Issue by tongnie** [snap-research/MLPInit-for-GNNs#2]
Hi! thanks for providing such an insightful paper! I'm curious about whether the MLPInit can deal with data with missing node features, which are common in practice. In this case, MLP may not be well trained since graph structure is unavailable. So how your method be applied in such case?

Thanks in advance for your help!

**Comment by ahxt** [comment]
Thank you for your question. Our method, as well as GNN, is probably infeasible for a graph without node features. However, an alternative approach for such a graph is to use pre-trained node embeddings as node features. In this case, our method would be feasible. (For node classification, the graph data typically includes node features, as far as I know.)

Thanks.

**Comment by tongnie** [comment]
Thanks for your prompt reply! I've run experiments with missing node features and try to train the GNN to reconstruct these missing values.  However, I found that GNN can perform well on this task, but without message passing, a pure MLP can not handle this. So in this case, the trained MLP could not serve as a good initialization for GNNs.

**Comment by ahxt** [comment]
Thank you for your exploration and experimental findings. The experimental results make sense to me. Since the pure MLP depends solely on node features, the quality of node features dominates the performance of MLP. Thank you for sharing your results

**Comment by tongnie** [comment]
Thanks for your sharings. There is one more question that still confuses me and I'd like to hear your advice. Does the performence of MLPInit depends on the downstream tasks? In your paper, you've found that the MLP and GNN could have the same feature space. Despite, I've found that their features could be different. In other words, if the task is to directly apply the extracted features of GNN to perform regression, without a sigmoid or softmax function, then the performance of pure MLP may be inferior since the output features are different from GNNs'.  How does MLP serve as a good initialization for GNN in this case? 


Look forward to your reply!

**Comment by ahxt** [comment]
Thanks for your question. I would say that if you would like to apply the extracted features from a GNN to perform a downstream task, PeerMLP cannot be used in this case, as PeerMLP requires the input feature and hidden layers of the GNN to have the same dimension.
